sidestep_engine/core/trainer_helpers.py

```
@@ -216,14 +216,20 @@ def capture_rng_state(device: Any = None) -> Dict[str, Any]:
+        if torch.cuda.is_available():
             rng["cuda_device"] = idx
             rng["cuda_rng"] = torch.cuda.get_rng_state(idx)
-        except Exception as exc:
-            logger.debug("Could not capture CUDA RNG state: %s", exc)
+        elif torch.mps.is_available():
+            rng["mps_device"] = idx
+            rng["mps_rng"] = torch.mps.get_rng_state(dev)
+        elif torch.xpu.is_available():
+            rng["xps_device"] = idx
+            rng["xps_rng"] = torch.mps.get_rng_state(dev)
+    except Exception as exc:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)